easybuild/tools/modules.py

```diff
@@ -555,11 +555,21 @@ def mod_exists_via_show(mod_name):
             # - Check first non-whitespace line for something that looks like an absolute path terminated by a colon
             mod_exists_regex = r'\s*/.+:\s*'
             for line in stderr.split('\n'):
+                # skip whitespace lines
+                self.log.debug("Checking line '%s' to determine whether %s exists...", line)
                 if OUTPUT_MATCHES['whitespace'].search(line):
+                    self.log.debug("Treating line '%s' as whitespace, so skipping it", line)
                     continue
-                if OUTPUT_MATCHES['error'].search(line):
+                # skip lines that start with 'module-version',
+                # see https://github.com/easybuilders/easybuild-framework/issues/3376
+                elif line.startswith('module-version '):
```

Comment:
Doesn't ModuleTCL also have things like `module-alias` or so? They should be handled the same, so I'd propose `line.startswith('module-'):` here.

I'd also move this check after the error check so we don't accidentally pick up an error. I think the whitespace and error regexps are a good safe guard before any "manual evalulation".

I also have mixed feelings about the `elif`. It doesn't mix well with the `continue` and `break`. Maybe remove the `continue` and make all a "real` `if-elif-else` chain? I guess the last one can be: `else: res = re.match(mod_exists_regex, line); break`

Reply:
I prefer keeping the `continue` and `break`, it helps making to logic more clear, so I ditched the `elif` again, see f237afe